pkg/kubeadm/master/addons.go

```diff
@@ -223,11 +230,15 @@ func createKubeDNSServiceSpec(s *kubeadmapi.KubeadmConfig) (*api.ServiceSpec, er
 }
 
 func CreateEssentialAddons(s *kubeadmapi.KubeadmConfig, client *clientset.Clientset) error {
-	kubeProxyDaemonSet := NewDaemonSet(kubeProxy, createKubeProxyPodSpec(s))
-	SetMasterTaintTolerations(&kubeProxyDaemonSet.Spec.Template.ObjectMeta)
+	arches := [3]string{"amd64", "arm", "arm64"}
+
+	for _, arch := range arches {
```

Comment:
I'm completely on board with adding multi-arch capability from the start, as it doesn't cost us much to have. However, for we would need to ensure it's only done for the proxy. Would you mind to remove the loop, and simply ensure that proxy (and other resources) have architecture node labels set for `runtime.GOARCH`. Initially, you can simply deploy the extra copies of your favourite archs after the fact using `kubectl`... Would this be reasonable thing to agree on from your perspective? Follow-up PR agains master to enable full-blow multi-architecture add-ons would be fine, but only after we get the MVP into master.